File: golang/codec/message_openconfirm.go
package codec

import (
	"encoding/binary"
	"fmt"
)

type OpenConfirmMessage struct {
	ChannelID     uint32
	SenderID      uint32
	WindowSize    uint32
	MaxPacketSize uint32
}

func (msg OpenConfirmMessage) String() string {
	return fmt.Sprintf("{OpenConfirmMessage ChannelID:%d SenderID:%d WindowSize:%d MaxPacketSize:%d}",
		msg.ChannelID, msg.SenderID, msg.WindowSize, msg.MaxPacketSize)
}

func (msg OpenConfirmMessage) Channel() (uint32, bool) {
	return msg.ChannelID, true
}

func (msg OpenConfirmMessage) MarshalMux() ([]byte, error) {
	packet := make([]byte, payloadSizes[msgChannelOpenConfirm]+1)
	packet[0] = msgChannelOpenConfirm
	binary.BigEndian.PutUint32(packet[1:5], msg.ChannelID)
	binary.BigEndian.PutUint32(packet[5:9], msg.SenderID)
	binary.BigEndian.PutUint32(packet[9:13], msg.WindowSize)
	binary.BigEndian.PutUint32(packet[13:17], msg.MaxPacketSize)
	return packet, nil
}

func (msg *OpenConfirmMessage) UnmarshalMux(b []byte) error {
	msg.ChannelID = binary.BigEndian.Uint32(b[1:5])
	msg.SenderID = binary.BigEndian.Uint32(b[5:9])
	msg.WindowSize = binary.BigEndian.Uint32(b[9:13])
	msg.MaxPacketSize = binary.BigEndian.Uint32(b[13:17])
	return nil
}

File: csharp/qmux/codec/message_openconfirm.cs
namespace qmux.codec;

using System;
using System.IO;
using System.Text;

public struct OpenConfirmMessage : IMessage
{
    public UInt32 ChannelId;
    public UInt32 SenderId;
    public UInt32 WindowSize;
    public UInt32 MaxPacketSize;

    public string String()
    {
        return $"OpenConfirmMessage ChannelId:{this.ChannelId} SenderId:{this.SenderId} WindowSize:{this.WindowSize} MaxPacketSize:{this.MaxPacketSize}";
    }
    public (UInt32, bool) Channel()
    {
        return (this.ChannelId, true);
    }

    public byte[] MarshalMux()
    {
        // 16 + 1
        var bufferSize = (int)PayloadSizes.MessageChannelOpenConfirm + 1;
        var stream = new MemoryStream(new byte[bufferSize]);
        using (var writer = new BinaryWriter(stream, Encoding.BigEndianUnicode, false))
        {
            // 0
            writer.Write((byte)MessageType.MessageChannelOpenConfirm);
            // 1-5
            writer.Write(this.ChannelId);
            // 5-9
            writer.Write(this.SenderId);
            writer.Write(this.WindowSize);
            writer.Write(this.MaxPacketSize);
        }
        return stream.ToArray();
    }
    public void UnmarshalMux(byte[] b)
    {
        var stream = new MemoryStream(b);
        using (var reader = new BinaryReader(stream, Encoding.BigEndianUnicode))
        {
            // 0
            reader.ReadByte();
            // 1-5
            this.ChannelId = reader.ReadUInt32();
            this.SenderId = reader.ReadUInt32();
            this.WindowSize = reader.ReadUInt32();
            this.MaxPacketSize = reader.ReadUInt32();
        }
    }
}